### PR TITLE
move dolt plugin dev_setup.sh script logic into pytest fixture

### PR DESCRIPTION
**Pull request by cosmicBboy**
Signed-off-by: cosmicBboy <niels.bantilan@gmail.com>

# TL;DR

This PR makes the dolt plugin tests more stable. Before this fix, running it locally on my system raises this configuration error:

```
_____________________________________ ERROR at setup of test_dolt_table_write ______________________________________
repo_dir = '/var/folders/4q/frdnh9l10h53gggw1m59gr9m0000gn/T/tmpwt7z0aqy/foo', error = False
    @staticmethod
    def init(repo_dir: Optional[str] = None, error: bool = False) -> "Dolt":
        """
        Creates a new repository in the directory specified, creating the directory if `create_dir` is passed, and returns
        a `Dolt` object representing the newly created repo.
        :return:
        """
        if not repo_dir:
            repo_dir = os.getcwd()
        os.makedirs(repo_dir, exist_ok=True)
        logger.info(f"Initializing Dolt repo in {repo_dir}")
        try:
>           _execute(["init"], cwd=repo_dir)
../../miniconda3/envs/flytekit/lib/python3.7/site-packages/doltcli/dolt.py:357:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
args = ['init'], cwd = '/var/folders/4q/frdnh9l10h53gggw1m59gr9m0000gn/T/tmpwt7z0aqy/foo', outfile = None
    def _execute(args: List[str], cwd: Optional[str] = None, outfile: Optional[str] = None):
        from .utils import DOLT_PATH
        _args = [DOLT_PATH] + args
        str_args = " ".join(" ".join(args).split())
        logger.info(str_args)
        if outfile:
            with open(outfile, "w") as f:
                # _outfile = open(outfile, "w") if outfile else PIPE
                proc = Popen(args=_args, cwd=cwd, stdout=f, stderr=PIPE)
        else:
            proc = Popen(args=_args, cwd=cwd, stdout=PIPE, stderr=PIPE)
        out, err = proc.communicate()
        exitcode = proc.returncode
        if exitcode != 0:
            logger.error(err)
>           raise DoltException(str_args, out, err, exitcode)
E           doltcli.dolt.DoltException: ('init', b'\nSpecific dolt init options\n    --name=<name>\n      The name used in commits to this repo. If not provided will be taken from user.name in the global config.\n    \n    --email=<email>\n      The email address used. If not provided will be taken from user.email in the global config.\n    \n    --date=<date>\n      Specify the date used in the initial commit. If not specified the current system time is used.\n    \n', b'Could not determine user.name. Use the init parameter --name "FIRST LAST" to set it for this repo, or dolt config --global --add user.name "FIRST LAST"\n', 1)
../../miniconda3/envs/flytekit/lib/python3.7/site-packages/doltcli/dolt.py:91: DoltException
During handling of the above exception, another exception occurred:
doltdb_path = '/var/folders/4q/frdnh9l10h53gggw1m59gr9m0000gn/T/tmpwt7z0aqy/foo'
    @pytest.fixture(scope="function")
    def db(doltdb_path):
        try:
>           db = dolt.Dolt.init(doltdb_path)
plugins/tests/dolt/test_wf.py:53:
```

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [ ] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: plugins/tests/dolt/test_wf.py [hunk 1]
```diff
@@ -20,16 +21,20 @@ def dolt_install():
...
+        dolt_path = os.path.join(tmp_dir.name, "dolt")
+        subprocess.run(install_script, env={"INSTALL_PATH": tmp_dir.name})
+        for attr, value in [("user.name", "First Last"), ("user.email", "first.last@example.com")]:
+            subprocess.run([dolt_path, "config", "--global", "--add", attr, value])
         dolt.set_dolt_path(dolt_path)
         yield dolt_path
     finally:
-        shutil.rmtree(d.name)
+        shutil.rmtree(tmp_dir.name)
+        shutil.rmtree(Path.home() / ".dolt")
```
Comment (max-hoffman) [inline plugins/tests/dolt/test_wf.py:37]: is this necessary? are we init'ing outside of the tempdir?

**Review by max-hoffman — APPROVED** [review]
lgtm, small comment

File: plugins/tests/dolt/test_wf.py (hunk 1 above)
Comment (cosmicBboy) [reply plugins/tests/dolt/test_wf.py:37]: at least locally the config options would be saved in `~/.dolt/config_global.json`, which would persist across invocations of `pytest`

Comment (max-hoffman) [reply plugins/tests/dolt/test_wf.py:37]: oh good catch. --local instead of --global is another option, this looks good though 👍
